app/src/main/java/bg/zahov/app/ui/history/calendar/CalendarScreen.kt

```diff
@@ -0,0 +1,181 @@
+        daysOfWeek = uiState.daysOfWeek,
+        numberOfWorkoutsPerMonth = uiState.numberOfWorkoutsPerMonth
+    )
+}
+
+@Composable
+fun CalendarContent(
+    calendarState: CalendarState,
+    daysOfWeek: List<DayOfWeek>,
+    dayToHasUserWorkedOut: Map<LocalDate, Boolean>,
+    numberOfWorkoutsPerMonth: Map<Month, String>
+) {
+    HorizontalCalendar(
+        state = calendarState,
+        dayContent = { Day(it, dayToHasUserWorkedOut[it.date] == true) },
+        monthHeader = {
+            MonthHeader(
+                text = it.yearMonth.month.name,
+                daysOfWeek = daysOfWeek
+            )
+        },
+        monthFooter = {
+            MonthFooter(
+                workoutCount = stringResource(
+                    R.string.workouts_for_month,
+                    numberOfWorkoutsPerMonth[it.yearMonth.month] ?: 0
+                ),
+                year = it.yearMonth.year.toString(),
+            )
+        }
+    )
+}
+
+@Composable
+fun Day(
+    day: CalendarDay, hasUserWorkedOut: Boolean = true,
+    boxModifier: Modifier = Modifier
+        .aspectRatio(1f)
+        .background(
+            color = colorResource(R.color.background)
+        )
+        .padding(4.dp)
+) {
+    Box(
+        modifier = boxModifier,
+        contentAlignment = Alignment.Center
+    ) {
+        Text(
+            text = day.date.dayOfMonth.toString(),
+            color = if (day.position == DayPosition.MonthDate) Color.White else Color.Gray
+        )
+        if (hasUserWorkedOut) {
+            Icon(
+                painter = painterResource(R.drawable.ic_check_mark),
+                contentDescription = "",
+                modifier = Modifier
+                    .size(16.dp)
+                    .align(Alignment.TopEnd),
+                tint = Color.Unspecified
+            )
+        }
+    }
+}
+
+@Composable
+fun MonthHeader(
+    columnModifier: Modifier = Modifier.padding(top = 24.dp, bottom = 24.dp),
+    textModifier: Modifier = Modifier.padding(top = 8.dp, bottom = 4.dp),
+    daysModifier: Modifier = Modifier
+        .padding(top = 8.dp)
+        .fillMaxWidth(),
+    text: String,
+    daysOfWeek: List<DayOfWeek> = listOf()
+) {
+    Column(modifier = columnModifier, horizontalAlignment = Alignment.CenterHorizontally) {
+        Text(
+            modifier = textModifier,
+            text = text,
+            style = MaterialTheme.typography.titleLarge,
+            fontWeight = FontWeight.Bold,
+            color = Color.White
+        )
+
+        LazyRow(
+            modifier = daysModifier,
+            userScrollEnabled = false,
+            horizontalArrangement = Arrangement.SpaceAround
+        ) {
+            items(count = daysOfWeek.size) { pos ->
+                MonthHeaderDayText(Modifier.weight(1f), daysOfWeek[pos])
+            }
+        }
+    }
+}
+
+@Composable
+fun MonthHeaderDayText(modifier: Modifier, day: DayOfWeek) {
+    Text(
```

Comment:
don't forget to put Modifier: Modifier = Modifier, in order to extend you modifiers when you call a composable

Reply:
👍